### DISCUSSION TITLE
Alternate to spring.sleuth.baggage.remote-fields

### DISCUSSION THREAD

**Discussion by hshah2811** [signalfx/splunk-otel-java#1538] [Q&A]
We have removed spring cloud sleuth library and migrated to splunk Otel java instrumentation library to propagate traces between micro-services for distributed tracing (leveraging SignalFX).  Traces and spans are working fine. 

We also want to propagate baggage as custom header in our downstream services. 
Expectation: X-Correlation-ID:<value from upstream> will be available in header. My app should propagate the same header in every downstream services. (custom baggage propagation). 

Anyone can help how to achieve that?

**Comment by laurit** [comment]
Sorry for the late reply. I believe you can achieve this with a custom propagator, for inspiration see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/src/main/java/com/example/javaagent/DemoPropagator.java
